unity-renderer/Assets/DCLServices/EmotesCatalog/EmotesCatalogService.cs

```diff
@@ -31,11 +31,17 @@ public EmotesCatalogService(IEmotesCatalogBridge bridge, IAddressableResourcePro
 
     private async UniTaskVoid InitializeAsyncEmbeddedEmotes()
     {
+        bool taskCancelled = false;
+
         try
         {
             addressableCTS = new CancellationTokenSource();
             embeddedEmotesSO = await addressableResourceProvider.GetAddressable<EmbeddedEmotesSO>("EmbeddedEmotes.asset", addressableCTS.Token);
         }
+        catch (OperationCanceledException e)
+        {
+            taskCancelled = true;
+        }
```

Comment:
Another option, in order to avoid to create a new boolean variable, would be:

```
try
{
	...
}
catch (OperationCanceledException) { return; }
catch (Exception e)
{
	...
}

EmbedEmotes();
```

Reply:
AFAIK `finally` would be called regardless of the `return` in the `catch`

Reply:
Yes, for this reason I remove the `finally` block in my suggested alternative. Anyway, this was only another way to do the same as you already have. Feel free to use what you prefer.

Reply:
oops! sry, didn't notice that :)
I didn't want to remove the finally because of my lack of context regarding this class, but if you say I can remove it, I'll remove it 😎